plugins/bitbucket-cloud-common/scripts/adjust-models.js

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+
+const modelsFile = `${__dirname}/../src/models/index.ts`;
+
+let models = fs.readFileSync(modelsFile).toString();
+
+// the generator creates ${DefinitionName}AllOf interfaces for the "allOf" statements with a type definition
+// (as extension to another references one) as well as sub-interfaces or enums.
+// Those are duplicates of the interface ${DefinitionName}.
+const matches = models.matchAll(
+  /(?:interface|enum) (.*AllOf[^ \n]*)[ \n]+([^{]*)[{]/gm,
+);
+let match;
+// eslint-disable-next-line no-cond-assign
+while ((match = matches.next()).value) {
+  const interfaceOrEnumName = match.value[1];
+
+  const realInterfaceOrEnumName = interfaceOrEnumName.replace('AllOf', '');
+  const realMatch = models.match(
+    new RegExp(
+      `(?:interface|enum) ${realInterfaceOrEnumName}[ \n]+((?:[^{]|\n)*)[{]`,
+      'im',
+    ),
+  );
+
+  if (realMatch) {
+    // the generator creates ${DefinitionName}AllOf interfaces for the "allOf" with a type definition
+    // (as extension to another references one). Those are duplicates of the interface ${DefinitionName}
+    models = models.replaceAll(
+      new RegExp(
+        `\/[*].*\n(?: [*].*\n)* [*]\/\n.*${interfaceOrEnumName} .*\n(?: .*\n)*[}]\n\n`,
+        'gm',
+      ),
+      '',
+    );
+  } else {
+    models = models.replaceAll(interfaceOrEnumName, realInterfaceOrEnumName);
+  }
+}
+
+models = models
+  // change Paginated to use generics for the result items
+  .replace(
+    /interface Paginated( [{].*\n(?: .*\n)* +)values.*\n/m,
+    'interface Paginated<TResultItem>$1values?: Array<TResultItem> | Set<TResultItem>;\n',
+  )
+  // and let other Paginated[...] interface extend it.
+  // Those should nearly always only override the "values" property with the concrete type.
+  .replaceAll(
+    /extends Paginated( [{].*\n(?: .*\n)* +(?:readonly )?values.*<([^>]+))/gm,
+    'extends Paginated<$2>$1',
+  );
```

Comment:
this is all a bit magic :) hoping that the format doesn't change much down the line

Reply:
I added a bit more documentation